test_kcidb.py

```diff
@@ -429,3 +429,93 @@ def test_notify_main():
                               "Subject: Test build: .*\x00"
                               "Subject: Test test: .*\x00"
                               "Subject: Test test: .*\x00")
+
+
+def test_ingest_main():
+    """Check kcidb-ingest works"""
+    assert_executes('', "kcidb.ingest_main")
+    assert_executes('{', "kcidb.ingest_main",
+                    status=1, stderr_re=".*JSONParseError.*")
+    assert_executes('{}', "kcidb.ingest_main",
+                    status=1, stderr_re=".*ValidationError.*")
+    empty = json.dumps(dict(version=dict(major=SCHEMA.major,
+                                         minor=SCHEMA.minor)))
+    assert_executes(empty, "kcidb.ingest_main")
+    assert_executes(empty + empty, "kcidb.ingest_main")
+    git_commit_hash1 = "4ff6a2469104218a044ff595a0c1eb469ca7ea01"
+    git_commit_hash2 = "fe3fc1bc47d6333d7d06bc530c6e0c1044bab536"
+
+    one_of_everything = json.dumps(dict(
+        version=dict(major=SCHEMA.major, minor=SCHEMA.minor),
+        checkouts=[
+            dict(id="test:checkout:1",
+                 git_commit_hash=git_commit_hash1,
+                 patchset_hash="",
+                 origin="test")
+        ],
+        builds=[
+            dict(id="test:build:1",
+                 origin="test",
+                 checkout_id="test:checkout:1",
+                 valid=False)
+        ],
+        tests=[
+            dict(id="test:test:1",
+                 origin="test",
+                 build_id="test:build:1",
+                 waived=False,
+                 status="PASS")
+        ]
+    ))
```

Comment:
This is not really "everything" despite what the variable name says. We also have "issues" and "incidents" now, and it would be good to add those too. Look around the tests for sample data.

Reply:
@spbnick where I can find the tests for sample data?

Reply:
A command like this would have got you started: `grep -r --include 'test_*.py' incident`

Here's a good one: https://github.com/kernelci/kcidb/blob/c2de2921297d315117acdc1a3fd6305c27245e72/kcidb/test_monitor.py#L196-L231

But see the schema docs to figure out what means what.